crates/revm/src/optimism.rs

```diff
@@ -160,17 +151,8 @@ mod tests {
             l1_fee_scalar: U256::from(1_000),
         };
 
-        // The gas cost here should be zero since the tx is a deposit
         let input = bytes!("FACADE");
-        let gas_cost = l1_block_info.calculate_tx_l1_cost::<BedrockSpec>(&input, true);
-        assert_eq!(gas_cost, U256::ZERO);
-
-        let gas_cost = l1_block_info.calculate_tx_l1_cost::<RegolithSpec>(&input, false);
+        let gas_cost = l1_block_info.calculate_tx_l1_cost::<RegolithSpec>(&input);
         assert_eq!(gas_cost, U256::from(1048));
-
-        // Zero rollup data gas cost should result in zero for non-deposits
-        let input = bytes!("");
-        let gas_cost = l1_block_info.calculate_tx_l1_cost::<RegolithSpec>(&input, false);
-        assert_eq!(gas_cost, U256::ZERO);
```

Comment:
It's not explicitly undefined behavior to pass empty payloads into the `calculate_tx_l1_cost` function so I lean towards having the function return `U256::ZERO` if the tx is a deposit or the bytes are empty.

Then we can add these two test cases: input with empty bytes and input with the first byte as `0x7E`.

Reply:
done